lib/inc/drogon/HttpTypes.h

```diff
@@ -169,26 +170,37 @@ enum class WebSocketMessageType
     Close,
     Unknown
 };
-inline std::ostream &operator<<(std::ostream &out, drogon::ReqResult result)
+
+inline std::string toString(drogon::ReqResult result)
```

Comment:
@an-tao Personally I'd prefer to follow the STL using `to_string` as this may make some template easier to write. What's your thought?

Reply:
Yes, I agree with you, and what about using string_view instead? implement to_string and to_string_view?

Reply:
@an-tao @marty1885 
how about this?
![image](https://user-images.githubusercontent.com/11664992/147079523-2445d16d-a331-43b6-95ca-69eef1d75185.png)

Reply:
I don't think returning string_view is a good solution. Let's stick with STL convention.

I think we should directly output the string view instead of .data(). Please use drogon::string_view instead of the STL one. Drogon supports C++14 and drogon::string_view switches between STL and Boost depending on the current C++ version.

Reply:
@marty1885 
I 've push new commit.

> I don't think returning string_view is a good solution.
Can you explain more about this?

Reply:
Maybe I'm overthinking. And let me know if I am. My concern is that people will be using code in some sort, with the expectation that `to_string` will return a string due to it's name.

```c++
template <typename T>
void handle_error(const drogon::string_view source, const T& error_obj)
{
    std::string msg  = source + to_string(error_obj);
    ....
}
```

Which would need special handling if we return a string_view for ReqResult.

Reply:
I mean to_string returns std::string, and to_string_view returns drogon::string_view

Reply:
> Maybe I'm overthinking. And let me know if I am. My concern is that people will be using the following code in some sort, with the expectation that `to_string` will return a string due to it's name.

I think this argument is legit, and I’m absolutely with you on this one.

Reply:
so should we implement those 2 function

stringview for iostream, to_string like an alias to  stringview function?

Reply:
We should implement `std::string to_string(ReqResult)`, `drogon::string_view to_string_view(ReqResult)` and let `operator <<` output the string_view.